bin/mocha

```diff
@@ -16,7 +16,7 @@ var spawn = require('child_process').spawn,
 getOptions();
 
 process.argv.slice(2).forEach(function(arg){
-  var flag = arg.split('=')[0];
+  var flag = arg.split('=', 1)[0].replace(/([a-z])_/g, '$1-');
```

Comment:
I'm not really following, but please let me explain my intent:

1.  `node` only allows interchangeable `-` and `_` in those options listed within `--v8-options`.  For example, you cannot run `node --throw_deprecation`.
2.  This change, as proposed, would actually *add* support for cases such as `--throw_deprecation`.  This is not necessary, since `node` does not support it.
3.  Any modification of the command-line API without integration test coverage is risky.  We don't have a whole lot of coverage against the `mocha` executable in the first place (if any), and I'm almost certain we have *no* coverage for flags shown by `--v8-options`.  Due mainly to lack of integration tests (which would be a pain to write anyway), I'm not convinced that the proposed changes would not break something somebody is doing.
4.  So, adding the two lines as I proposed *should* add support either `-` or `_` for all of the `--v8-options`-type flags the `mocha` executable *currently* supports.  Since no lines are modified (and only added), this is relatively "safer".  Would it not address your use case?

To expand further, the `mocha` allows some--but not all--`node` flags through.  When a flag is added to `node`, someone invariably sends a PR which adds the flag to `mocha` as well.  This doesn't scale very well.  

The only reason (that I'm aware of) for providing *two* executables, `mocha` and `_mocha`, is to allow passing of `node` flags--merely a convenience.  All flags, of course, are supported by running `node [options] /path/to/_mocha`; `mocha` itself only supports a subset, as you can see.

I don't believe I have seen an issue come through where:
- someone needs support for, say, `--use-strict`
- cannot use `--use_strict`
- has no viable workaround

Because a workaround exists, and because the proposed changes are (in my mind, anyway) high-risk and low-reward, I can't really get behind changes to the documented API, nor modification of the option-parsing lines.